### PR TITLE
TELCODOCS-430: Support of multiple simultaneous patch (x.y.z) versions of Release/OS images

### PR DESCRIPTION
**Pull request by kdrake20**
Fixes: [TELCODOCS-430](https://issues.redhat.com/browse/TELCODOCS-430)

For: Version 4.11 

Signed-off-by: Katie Tothill ktothill@redhat.com

### CODE REVIEW

**Comment by ktothill** [comment]
**Who:** Daniel Erez,SME
**Contact Method:** Slack, Direct Message
**Date and Time:** June 6, 2022  15:32
**Summary:** Requested SME review. Reached out to Emily to identify QE and confirm if they're required, since this isn't a code change.

**Next Steps:** Review Daniel's suggestions and request QE review.

File: modules/ztp-creating-siteconfig-custom-resources.adoc [hunk 1]
```diff
@@ -71,8 +71,7 @@ spec:
    releaseImage: quay.io/openshift-release-dev/ocp-release:4.10.0-x86_64 <2>
 ----
 <1> The descriptive version that you want to deploy.
-<2> Points to the specific release image to deploy.
-
+<2> Specifies the `releaseImage` to deploy and determines the OS Image. The discovery ISO bases the releaseImage on the specified image or the latest version if the specified image is unavailable.
```
Comment (danielerez) [inline modules/ztp-creating-siteconfig-custom-resources.adoc:74]: Just the make it a clearer and more explicit, consider something like this rephrasing:
"Specifies the `releaseImage` to deploy and determines the version of OS image accordingly. I.e. The discovery ISO will be based on an OS image version as the specified `releaseImage`, or latest if the exact version is unavailable."

Hope it describes the service logic well enough, just a recap for the detailed behaviour:
* Extract the x.y.z version from `releaseImage` (e.g. 4.10.1)
* Search for OS image with the exact version x.y.z (e.g. 4.10.1)
* If not found, search for the latest z-stream (e.g. 4.10.z)
* Use the OS image to create the discovery ISO

**Comment by ktothill** [comment]
**Who:** Chad Crum, QE
**Contact Method:** Slack, Direct Message
**Date and Time:** June 7, 2022  09:48
**Summary:** Confirmed with Emily that Chad is the QE for this doc set. Requested QE review. 

**Next Steps:** Review Chad's suggestions and request peer reviews.

**Comment by ktothill** [comment]
**Contact Method:** Slack, [#telco-docs](https://coreos.slack.com/archives/C02GSQ4CSTE/p1654609940681329) & [#forum-docs-review](https://coreos.slack.com/archives/C025XCRRH28/p1654609974736529)
**Date and Time:** June 7, 2022  09:53
**Summary:** Requested reviews from Telco peers and Peer Review Squad

**Next Steps:** Review and implement feedback as needed.

File: modules/ztp-creating-siteconfig-custom-resources.adoc [hunk 2]
```diff
@@ -71,8 +71,7 @@ spec:
    releaseImage: quay.io/openshift-release-dev/ocp-release:4.10.0-x86_64 <2>
 ----
 <1> The descriptive version that you want to deploy.
-<2> Points to the specific release image to deploy.
-
+<2> Specifies the `releaseImage` to deploy and determines the OS Image version. The discovery ISO is based on an OS image version as the specified `releaseImage`, or latest if the exact version is unavailable.
```
Comment (kquinn1204) [inline modules/ztp-creating-siteconfig-custom-resources.adoc]: maybe a small nit but is the second "specified" needed. "The discovery ISO is based on an OS image version as the  `releaseImage`, or latest if the exact version is unavailable."

Comment (ktothill) [reply modules/ztp-creating-siteconfig-custom-resources.adoc]: Updated as suggested.

**Comment by ktothill** [comment]
Who: Chad Crum, QE
Contact Method: Slack, Direct Message
Date and Time: June 7, 2022 13:29
Summary: Requested @chadcrum's review of Release Notes:

> Added imageStorage parameter to agent_service_config.yaml file.

Next Steps: Review feedback and implement if needed.

**Comment by ktothill** [comment]
Contact Method: Slack, [#forum-docs-review](https://coreos.slack.com/archives/C025XCRRH28/p1654706990797089)
Date and Time: June 8, 2022  12:50
Summary: Requested merge in #forum-docs-review

**Comment by ktothill** [comment]
See [PR 46708](https://github.com/openshift/openshift-docs/pull/46708) for RN.

**Comment by jeana-redhat** [comment]
/cherrypick enterprise-4.10
Per Slack, this is also going into 4.10 in a z-stream

**Pull request merged**